packages/adapter-node/src/index.js

```diff
@@ -42,19 +42,26 @@ if (socket_activation) {
 	});
 }
 
-function shutdown() {
+/** @param {'SIGINT' | 'SIGTERM' | 'IDLE'} reason */
+function shutdown(reason) {
 	if (shutdown_timeout_id) return;
 
 	// @ts-expect-error this was added in 18.2.0 but is not reflected in the types
 	server.server.closeIdleConnections();
 
-	server.server.close(() => {
+	server.server.close((error) => {
+		// occurs if the server is already closed
+		if (error) return;
+
 		if (shutdown_timeout_id) {
 			shutdown_timeout_id = clearTimeout(shutdown_timeout_id);
 		}
 		if (idle_timeout_id) {
 			idle_timeout_id = clearTimeout(idle_timeout_id);
 		}
+
+		// @ts-expect-error custom events cannot be typed
+		process.emit('sveltekit:shutdown', reason);
```

Comment:
I'm a bit torn as to whether this is the right API. It is more convenient to work with but kind of duplicating/aggregating existing events. The alternative would be to have the user do this:
```
process.on('SIGINT', shutdownHandler)
process.on('SIGTERM', shutdownHandler)
process.on('sveltekit:idle', shutdownHandler)
```

I think what you're doing is probably nicer, but I thought I'd note it for posterity or in case other reviewers have a different opinion

Reply:
The problem is that this doesn't work with the graceful shitdown. If the user listens for the `SIGINT`and `SIGTERM` events to run some cleanup code then that cleanup code will run at the same time as the server is shutting down. But ideally you want to close the db connection etc. after the server has closed.

Reply:
Yeah makes sense. I just renamed it.